src/configuration/datastore.cr

```diff
@@ -1,11 +1,14 @@
 require "yaml"
+require "./s3"
 
 class Configuration::Datastore
   include YAML::Serializable
 
   getter mode : String = "etcd"
   getter external_datastore_endpoint : String = ""
 
+  getter s3 : Configuration::S3 = Configuration::S3.new
```

Comment:
By convention, I'd add an intermediate namespace as in `Configuration::Datastore::S3` since this configuration is specific to the datastore only.

Reply:
The hierarchy of the classes looks much better now 👍